packages/mobile/src/navigator/Navigator.tsx

```diff
@@ -527,18 +499,81 @@ const modalScreenOptions: ScreenOptions = Platform.select({
   }),
 })
 
+const animatedScreens = (Navigator: typeof Stack) => (
```

Comment:
Is there an approach that doesn't require us to group together these unrelated screens this way? I'm worried that this grouping could lead to navigation bugs. I don't have a particular case in mind but something like a back button not working as expected for screens in a Stack Navigator

Reply:
I feel what you say, but unfortunately, I don't see another way for now.
`react-navigation` mentions we should avoid unnecessary nesting and only use them to get the desired UI (tab bar, modal, etc)

> Think of nesting navigators as a way to achieve the UI you want rather than a way to organize your code.

See https://reactnavigation.org/docs/nesting-navigators/#best-practices-when-nesting